src/umpire/ResourceManager.cpp

```diff
@@ -495,22 +495,24 @@ void* ResourceManager::reallocate(void* current_ptr, std::size_t new_size)
 {
   strategy::AllocationStrategy* strategy;
 
-  UMPIRE_REPLAY(R"( "event": "reallocate", "payload": {)"
-                << R"( "ptr": ")" << current_ptr << R"(")"
-                << R"(, "size": )" << new_size << R"( } )");
-
   if (current_ptr != nullptr) {
     auto alloc_record = m_allocations.find(current_ptr);
     strategy = alloc_record->strategy;
   } else {
     strategy = getDefaultAllocator().getAllocationStrategy();
   }
 
+  UMPIRE_REPLAY(R"( "event": "reallocate", "payload": {)"
+                << R"( "ptr": ")" << current_ptr << R"(")"
+                << R"(, "size": )" << new_size << R"(, "allocator_ref": ")"
+                << strategy << R"(" } )");
+
```

Comment:
Note: @davidbeckingsale, I made a change to have Umpire also log the allocator_ref that was used for the reallocation.  This enables the replay tool to track reallocations correctlys.